core/key_test.go

```diff
@@ -159,157 +162,159 @@ func TestNewIndexKeyFromString_IfFullKeyString_ReturnKey(t *testing.T) {
 	assert.Equal(t, "idx", key.IndexName)
 }
 
-func toFieldValues(values ...string) [][]byte {
-	var result [][]byte = make([][]byte, 0, len(values))
-	for _, value := range values {
-		result = append(result, []byte(value))
+func encodePrefix(colID, indexID uint32) []byte {
+	return encoding.EncodeUvarintAscending(append(encoding.EncodeUvarintAscending(
+		[]byte{'/'}, uint64(colID)), '/'), uint64(indexID))
+}
+
+func encodeKey(colID, indexID uint32, fieldParts ...any) []byte {
+	b := encodePrefix(colID, indexID)
+	const partSize = 4
+	if len(fieldParts)%partSize != 0 {
+		panic(fmt.Sprintf("fieldParts must be a multiple of %d: id, kind, value, descending", partSize))
+	}
+	for i := 0; i < len(fieldParts)/partSize; i++ {
+		b = append(b, '/')
+		b = encoding.EncodeUvarintAscending(b, uint64(fieldParts[i*partSize].(int)))
+		b = encoding.EncodeUvarintAscending(b, uint64(fieldParts[i*partSize+1].(client.FieldKind)))
+		isDescending := fieldParts[i*partSize+3].(bool)
+		if fieldParts[i*partSize+2] == nil {
+			if isDescending {
+				b = encoding.EncodeNullDescending(b)
+			} else {
+				b = encoding.EncodeNullAscending(b)
+			}
+		} else {
+			if isDescending {
+				b = encoding.EncodeUvarintDescending(b, uint64(fieldParts[i*partSize+2].(int)))
+			} else {
+				b = encoding.EncodeUvarintAscending(b, uint64(fieldParts[i*partSize+2].(int)))
+			}
+		}
 	}
-	return result
+	return b
 }
 
-func TestIndexDatastoreKey_ToString(t *testing.T) {
+const intKind = client.FieldKind_NILLABLE_INT
+
+func TestIndexDatastoreKey_Bytes(t *testing.T) {
 	cases := []struct {
-		Key      IndexDataStoreKey
-		Expected string
+		Name        string
+		Key         IndexDataStoreKey
+		Expected    []byte
+		ExpectError bool
```

Comment:
I don't use the name as the error message. The last parameter to `assert` functions is additional info for output in case of failed test.
In this test the expected error is supposed to be returned by `encoding` package, which already have tests with expectatoion on specific error messages. Here we just want to make sure an error is returned.